File: src/Walrhouse.Application/Items/Queries/GetItems/ItemDto.cs
using Walrhouse.Domain.Entities;
using Walrhouse.Domain.Enums;

namespace Walrhouse.Application.Items.Queries.GetItems;

public class ItemDto
{
    public string ItemCode { get; set; } = string.Empty;
    public string ItemName { get; set; } = string.Empty;
    public int UoMGroupId { get; set; }
    public string BarcodeValue { get; set; } = string.Empty;
    public BarcodeFormat? BarcodeFormat { get; set; }
    public ItemGroup? ItemGroup { get; set; }
    public string? Remarks { get; set; }
}

File: src/Walrhouse.Web/App/WebClient/src/features/inventory/types/item-dto.ts
import { BarcodeFormat, ItemGroup } from '@/features/item/types';

export interface ItemDto {
  itemCode: string;
  itemName: string;
  uoMGroupId: number | null;
  barcodeValue?: string;
  barcodeFormat?: BarcodeFormat | null;
  itemGroup?: ItemGroup | null;
  remarks?: string | null;
}
